.changeset/short-coins-bow.md

```diff
@@ -0,0 +1,17 @@
+---
+'vscode-mdx': minor
+---
+
+Support for highlighting JSX evaluated expressions
+
+In JSX, you can include JS expressions within tags, like:
+```
+<ATag>
+{doSomething(‘a’, 7)}
+</asdfadf
+```
+
+This PR adds syntax highlighting for those expressions by adding a new pattern in the `tmLanguage.json`.
+
+### Limitations
+It will still highlight JS that isn’t valid to include in a JSX evaluated expression (eg. a class definition), which as far as I can tell is a limitation of how embedded languages work in TextMate syntax highlighting definitions.
```

Comment:
It’s totally valid to include a class declaration in a JSX expression!

Some things are invalid though, e.g. ESM. I don’t think that’s a limitation though. The ability to provide syntax highlighting for invalid code is a feature, not a bug or limitation. Invalid syntax will be reported by the intellisense introduced in #226.

```suggestion
```